src/main/java/com/juniorstart/juniorstart/controller/UserTechnologyController.java

```diff
@@ -0,0 +1,39 @@
+package com.juniorstart.juniorstart.controller;
+
+import com.juniorstart.juniorstart.model.UserTechnology;
+import com.juniorstart.juniorstart.security.CurrentUser;
+import com.juniorstart.juniorstart.security.UserPrincipal;
+import com.juniorstart.juniorstart.service.UserTechnologyService;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+import java.util.Set;
+
+@RestController
+@RequestMapping("/user/technology")
+public class UserTechnologyController {
+
+    private UserTechnologyService userTechnologyService;
+
+    public UserTechnologyController(UserTechnologyService userTechnologyService){
+        this.userTechnologyService = userTechnologyService;
+    }
+
+    @PreAuthorize("hasRole('USER')")
+    @PostMapping("")
+    public ResponseEntity<?> addUserProfileTechnology(@RequestBody Set<UserTechnology> userTechnology, @CurrentUser UserPrincipal userPrincipal){
+        return userTechnologyService.addUserTechnology(userTechnology, userPrincipal);
+    }
+
+    @PreAuthorize("hasRole('USER')")
+    @PutMapping("")
+    public ResponseEntity<?> updateUserProfileTechnology(@RequestBody UserTechnology userTechnology, @CurrentUser UserPrincipal userPrincipal){
+        return userTechnologyService.updateUserTechnology(userTechnology, userPrincipal);
+    }
+
+    @PreAuthorize("hasRole('USER')")
+    @DeleteMapping("")
+    public ResponseEntity<?> deleteUserProfileTechnology(@RequestBody UserTechnology userTechnology, @CurrentUser UserPrincipal userPrincipal){
+        return userTechnologyService.deleteUserTechnology(userTechnology, userPrincipal);
```

Comment:
To samo co w UserAdditionalSkillController

Reply:
Get jest wydzielony w B004, ale i tak muszę zmienić z racji dodatkowego Pola - AdditionalSkill, najchętniej bym zmienił kod, ale już w nowym tasku/branchu. Ale to już się dostosuję.